druid/src/contexts.rs

```diff
@@ -271,11 +271,16 @@ impl<'a> EventCtx<'a> {
 
     /// Request keyboard focus.
     ///
+    /// Calling this when the widget is already focused does nothing.
+    ///
     /// See [`has_focus`] for more information.
     ///
     /// [`has_focus`]: struct.EventCtx.html#method.has_focus
     pub fn request_focus(&mut self) {
-        self.base_state.request_focus = Some(FocusChange::Focus(self.widget_id()));
+        let id = self.widget_id();
+        if self.focus_widget.filter(|focus| focus == &id) == None {
```

Comment:
not a blocker, but I might write this as `if Some(self.widget_id()) == self.focus_widget {`?

Reply:
... or we can wait until we have `is_focused`.

Reply:
You mean `if Some(self.widget_id()) != self.focus_widget {` I presume, because we want to request focus when `self.focus_widget` is either `None` or `Some(other_id)`.